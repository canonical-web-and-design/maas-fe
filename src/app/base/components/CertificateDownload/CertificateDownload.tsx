import {
  Button,
  CodeSnippet,
  Icon,
  Textarea,
} from "@canonical/react-components";
import fileDownload from "js-file-download";

import type { CertificateData } from "app/store/general/types";

type Props = {
  certificate: CertificateData["certificate"];
  filename: string;
  isGenerated?: boolean;
};

const CertificateDownload = ({
  certificate,
  filename,
  isGenerated = false,
}: Props): JSX.Element => {
  return (
    <>
      {isGenerated ? (
        <div className="certificate-download">
          <CodeSnippet
            blocks={[
              { code: "lxc config trust add - <<EOF" },
              { code: certificate },
              { code: "EOF" },
            ]}
            className="u-no-margin--bottom"
            data-testid="certificate-code-snippet"
          />
        </div>
      ) : (
        <Textarea
          className="p-textarea--readonly"
          data-testid="certificate-textarea"
          id="lxd-cert"
          readOnly
          rows={5}
          value={certificate}
        />
      )}
      <Button
        data-testid="certificate-download-button"
        onClick={() => {
          fileDownload(certificate, filename);
        }}
        type="button"
      >
        Download certificate
        <span className="u-nudge-right--small">
          <Icon name="begin-downloading" />
        </span>
      </Button>
    </>
  );
};

export default CertificateDownload;
