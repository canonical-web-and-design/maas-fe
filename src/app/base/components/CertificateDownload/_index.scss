@mixin CertificateDownload {
  .certificate-download {
    @extend %vf-is-bordered;

    margin-bottom: $spv--large;
    max-height: 15rem;
    overflow-y: auto;
  }
}
