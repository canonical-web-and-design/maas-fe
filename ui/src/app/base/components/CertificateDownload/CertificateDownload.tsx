import { Button, CodeSnippet, Icon } from "@canonical/react-components";
import fileDownload from "js-file-download";

import type { CertificateData } from "app/store/general/types";

type Props = {
  certificate: CertificateData["certificate"];
  filename: string;
};

const CertificateDownload = ({ certificate, filename }: Props): JSX.Element => {
  return (
    <>
      <div className="certificate-download">
        <CodeSnippet
          blocks={[
            { code: "lxc config trust add - <<EOF" },
            { code: certificate },
            { code: "EOF" },
          ]}
          className="u-no-margin--bottom"
        />
      </div>
      <Button
        data-test="certificate-download-button"
        onClick={() => {
          fileDownload(certificate, filename);
        }}
        type="button"
      >
        Download certificate
        <span className="u-nudge-right--small">
          <Icon name="begin-downloading" />
        </span>
      </Button>
    </>
  );
};

export default CertificateDownload;
