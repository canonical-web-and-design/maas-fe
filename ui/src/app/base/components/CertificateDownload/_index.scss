@mixin CertificateDownload {
  .certificate-download {
    @extend %vf-is-bordered;

    margin-bottom: $spv-outer--medium;
    max-height: 15rem;
    overflow-y: auto;
  }
}