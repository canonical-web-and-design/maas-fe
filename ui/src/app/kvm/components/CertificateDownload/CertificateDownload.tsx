import { Button, CodeSnippet, Icon } from "@canonical/react-components";

type Props = {
  certificateString: string;
};

const CertificateDownload = ({ certificateString }: Props): JSX.Element => {
  return (
    <>
      <div className="certificate-download">
        <CodeSnippet
          blocks={[
            { code: "lxc config trust add - <<EOF" },
            { code: certificateString },
            { code: "EOF" },
          ]}
          className="u-no-margin--bottom"
        />
      </div>
      <Button
        onClick={() => {
          // TODO: Add download functionality
          // https://github.com/canonical-web-and-design/app-squad/issues/248
          return null;
        }}
        type="button"
      >
        Download certificate
        <span className="u-nudge-right--small">
          <Icon name="begin-downloading" />
        </span>
      </Button>
    </>
  );
};

export default CertificateDownload;
