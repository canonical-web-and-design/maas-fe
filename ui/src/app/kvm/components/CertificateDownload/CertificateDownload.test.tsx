import { shallow } from "enzyme";

import CertificateDownload from "./CertificateDownload";

describe("CertificateDownload", () => {
  it("renders", () => {
    const wrapper = shallow(
      <CertificateDownload certificateString="certificate" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
