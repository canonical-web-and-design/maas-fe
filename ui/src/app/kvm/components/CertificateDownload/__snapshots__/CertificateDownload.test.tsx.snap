// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificateDownload renders 1`] = `
<Fragment>
  <div
    className="certificate-download"
  >
    <CodeSnippet
      blocks={
        Array [
          Object {
            "code": "lxc config trust add - <<EOF",
          },
          Object {
            "code": "certificate",
          },
          Object {
            "code": "EOF",
          },
        ]
      }
      className="u-no-margin--bottom"
    />
  </div>
  <Button
    onClick={[Function]}
    type="button"
  >
    Download certificate
    <span
      className="u-nudge-right--small"
    >
      <Icon
        name="begin-downloading"
      />
    </span>
  </Button>
</Fragment>
`;
